js/ad_manager_vast.js

```diff
@@ -369,16 +369,24 @@ OO.Ads.manager(function(_, $) {
 
       // Save the stream data for use by VideoController
       var streams = {};
+      var streamData = null;
       var linearAd = ad.data.linear;
       if (linearAd && linearAd.mediaFiles) {
         var vastStreams = linearAd.mediaFiles;
-        streams["webm"] = this._extractStreamForType(linearAd.mediaFiles, "webm");
-        if (ad.streamUrl == null) {
-          streams["mp4"] = this._extractStreamForType(linearAd.mediaFiles, "mp4");
-          OO.log("extract m3u8 stream here"); // TODO
+        var videoEncodingsSupported = OO.VIDEO.ENCODING;
+        for (var encoding in videoEncodingsSupported) {
+          streamData = null;
+          if (videoEncodingsSupported.hasOwnProperty(encoding)) {
+            streamData = this._extractStreamForType(vastStreams, videoEncodingsSupported[encoding]);
+          }
+          if (streamData) {
+            streams[videoEncodingsSupported[encoding]] = streamData;
+            if (ad.streamUrl == null || videoEncodingsSupported[encoding] == OO.VIDEO.ENCODING.MP4) {
+              ad.streamUrl = streams[videoEncodingsSupported[encoding]];
```

Comment:
I see.  I think since the VTC is selecting the stream this value could end up incorrect since the user may not favor mp4 (mp4 is not the top of the list by default).  I would suggest either reading the stream url from the videoControllerVideoElementCreated event in the AMC and passing it along to the ad object or omitting it entirely.  Might rather not have data than have wrong data